log/pretty/pretty.go

```diff
@@ -0,0 +1,41 @@
+package pretty
+
+import (
+	"bufio"
+	"io"
+
+	"github.com/hokaccha/go-prettyjson"
+)
+
+// NewPrettyWriter wraps the supplied output writer with a filter that converts lines of JSON to pretty-printed lines
+// containing syntax highlighting as formatted by the go-prettyjson library. If there are errors in converting the json
+// it is simply output as-is. Note, there is a line length limit in the buffer
+// (currently 64k defined by https://pkg.go.dev/bufio#pkg-constants)
+func NewPrettyWriter(out io.Writer) io.WriteCloser {
+	r, wr := io.Pipe()
+
+	go func() {
+		defer wr.Close()
+		defer r.Close()
+
+		br := bufio.NewScanner(r)
+		for br.Scan() {
+			raw := []byte(br.Text())
+			output, err := prettyjson.Format(raw)
+			if err != nil {
+				_, err = out.Write(append(raw, '\n'))
+				if err != nil {
+					panic("could not output raw line")
```

Comment:
This is very tricky. This PrettyWriter is only provided so that when running apps on our local dev environments we can get colourised and formatted output by using HUMAN_LOG. This would never run in production. In order to achieve this simply I used piped io within a separate goroutine. To pass back errors to the app would require channels and a change to the api making upgrading apps much more difficult for little reasonable benefit. I took the pragmatic approach, because the correct alternative would be to use an alternative slog.Handler ([see Log Formatting here](https://go.dev/wiki/Resources-for-slog)) which would then interfere with the use of the replaceAttr used by the standard JSONHandler. The effort involved in this didn't seem worth it just for local development running.